py_gtfs_rt_ingestion/lib/postgres_utils.py

```diff
@@ -1,11 +1,71 @@
 import os
+import platform
+from typing import Any, Tuple, Dict
 import urllib.parse as urlparse
+from multiprocessing import Process, Queue
 
 import boto3
 import sqlalchemy as sa
+from sqlalchemy.orm import sessionmaker
+from sqlalchemy.ext.declarative import declarative_base
 
 from .logging_utils import ProcessLogger
 
+SqlBase: Any = declarative_base()
+
+
+class MetadataLog(SqlBase):  # pylint: disable=too-few-public-methods
+    """Table for keeping track of parquet files in S3"""
+
+    __tablename__ = "metadataLog"
+
+    pk_id = sa.Column(sa.Integer, primary_key=True)
+    processed = sa.Column(sa.Boolean, default=sa.false())
+    path = sa.Column(sa.String(256), nullable=False, unique=True)
+    created_on = sa.Column(
+        sa.DateTime(timezone=True), server_default=sa.func.now()
+    )
+
+
+def get_db_password() -> str:
+    """
+    function to provide rds password
+
+    used to refresh auth token, if required
+    """
+    db_password = os.environ.get("DB_PASSWORD", None)
+    db_host = os.environ.get("DB_HOST")
+    db_port = os.environ.get("DB_PORT")
+    db_user = os.environ.get("DB_USER")
+    db_region = os.environ.get("DB_REGION", None)
+
+    if db_password is None:
+        # generate aws db auth token
+        client = boto3.client("rds")
+        return urlparse.quote_plus(
+            client.generate_db_auth_token(
+                DBHostname=db_host,
+                Port=db_port,
+                DBUsername=db_user,
+                Region=db_region,
+            )
+        )
+
+    return db_password
+
+
+def postgres_event_update_db_password(
+    _: sa.engine.interfaces.Dialect,
+    __: Any,
+    ___: Tuple[Any, ...],
+    cparams: Dict[str, Any],
+) -> None:
+    """
+    update database passord on every new connection attempt
+    this will refresh db auth token passwords
+    """
+    cparams["password"] = get_db_password()
```

Comment:
do we have a ticket to unify the postgres utils across both repos?

Reply:
not that I know of